src/mesh/Router.cpp

```diff
@@ -210,14 +210,43 @@ ErrorCode Router::send(MeshPacket *p)
     if (p->which_payloadVariant == MeshPacket_decoded_tag) {
         ChannelIndex chIndex = p->channel; // keep as a local because we are about to change it
 
+
+
+#if defined(HAS_WIFI) || defined(PORTDUINO)
+        //check if we should send decrypted packets to mqtt
+
+        //truth table:
+        /* mqtt_server  mqtt_encryption_enabled should_encrypt
+         *    not set                        0              1
+         *    not set                        1              1
+         *        set                        0              0
+         *        set                        1              1
+         * 
+         * => so we only encrypt mqtt if they have a custom mqtt server AND mqtt_encryption_enabled is FALSE
+         */ 
+
+        bool shouldActuallyEncrypt = true;
+        if (*radioConfig.preferences.mqtt_server && !radioConfig.preferences.mqtt_encryption_enabled) {
+            shouldActuallyEncrypt = false;
```

Comment:
Our configuration settings should always default to false and we should avoid changing existing behavior where possible. 

Request here is to change this to mqtt_encryption_disabled where the default is 0.

Reply:
I would agree with this in general, but I have not found a single user who has managed to decrypt the encrypted messages.

Reply:
Totally happy to change it, but I'll just let this conversation play out for a little while first

Reply:
This is used by our existing MQTT public server and hub. It's also part of the spec:

https://meshtastic.org/docs/software/other/mqtt

If we change this behavior, the spec should also be updated.

Reply:
It won't affect the existing public server - the preference is ignored if the user has not set a mqtt_server (and thus connects to mqtt.meshtastic.org).  In that case it's encrypted anyway.

That docs page isn't very representative of the current implementation (looks more like a brainstorm?).  I'll have a go at adding a section that explains how it actually works right now

Reply:
@joshpirihi I totally missed that in my review. Both mqtt server and the new conf has to be set.

Thanks for hearing this out. PR is accepted.

Reply:
awesome! thanks for your feedback, always appreciated!